cpp/src/parquet/column_reader.cc

```diff
@@ -580,13 +580,20 @@ std::shared_ptr<Buffer> SerializedPageReader::DecompressIfNeeded(
     memcpy(decompressed, page_buffer->data(), levels_byte_len);
   }
 
-  // Decompress the values
-  PARQUET_ASSIGN_OR_THROW(
-      auto decompressed_len,
-      decompressor_->Decompress(compressed_len - levels_byte_len,
-                                page_buffer->data() + levels_byte_len,
-                                uncompressed_len - levels_byte_len,
-                                decompression_buffer_->mutable_data() + levels_byte_len));
+  // GH-31992: DataPageV2 may store only levels and no values
```

Comment:
Can we fix this in the Snappy decompressor instead?

Reply:
Yes, which one do you prefer?

Reply:
It will be useful to other callers if we fix this on the Snappy side.

Reply:
So should I handle the `0,0` case in snappy 🤔?

Reply:
> Ok, it's really a bug in parquet-java, because a 0-size buffer compressed to a 1-size buffer using Snappy.

I may not understand you correctly. The file with 0-sized snappy-compressed data is produced by parquet-mr: https://github.com/apache/arrow/issues/31992#issuecomment-2567972537

Reply:
Well, parquet-mr is parquet-java, right?

Reply:
> So should I handle the `0,0` case in snappy 🤔?

No, sorry. We should work around it in Parquet.

Reply:
Yes, I mean parquet-java is able to produce 0-sized compressed data as the example. How can I reproduce your case where 0 input is compressed to 1 byte?

Reply:
It's not my case, it's just what Snappy produces when you ask it to compress 0 byte.

Reply:
> it's just what Snappy produces when you ask it to compress 0 byte.

IMO that's:
1. "Compressed" Data is 0 byte
2. Actually, the levels holds k bytes ?

I don't know how parquet-java works, I tried parquet rust and it failed to read

Reply:
Again, a 0-byte compressed data is invalid. It's probably a special case in the Parquet Java implementation.